lib/internal/modules/cjs/loader.js

```diff
@@ -218,6 +218,22 @@ function tryExtensions(p, exts, isMain) {
   return false;
 }
 
+// find the longest (possibly multi-dot) extension registered in
+// Module._extensions
+function findExtension(filename) {
+  const name = path.basename(filename);
+  let currentExtension;
+  let index;
+  let startIndex = 0;
+  while ((index = name.indexOf('.', startIndex)) !== -1) {
+    startIndex = index + 1;
+    if (index === 0) continue; // Skip dotfiles like .gitignore
```

Comment:
If we are to skip the dotfiles, shouldn't this just `break` out of here?

Reply:
@thefourtheye 
> If we are to skip the dotfiles, shouldn't this just break out of here?

No, because it's not skipping the whole dotfile, just the first dot of the dot file.
```js
path.extname(".dotfile")
// => ""
path.extname(".dotfile.ext")
// => ".ext"
```

Reply:
A test was added for this case.